numpyro/util.py

```diff
@@ -201,24 +202,40 @@ def progress_bar_factory(num_samples, num_chains):
 
     remainder = num_samples % print_rate
 
+    idx_map = {}
```

Comment:
could you add some comments here for how this works?

Reply:
added by _`calc_chain_idx`